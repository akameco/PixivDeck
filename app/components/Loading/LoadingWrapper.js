// @flow
import styled from 'styled-components';

const LoadingWrapper = styled.div`
	display: flex;
	justify-content: center;
	background: #eee;
	width: 100%;
	height: 100%;
`;

export default LoadingWrapper;
