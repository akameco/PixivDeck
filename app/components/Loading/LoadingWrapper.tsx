import styled from 'styled-components'
import { key } from 'styles/styleTheme'

const LoadingWrapper = styled.div`
  background-color: ${key('base')};
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100%;
`
export default LoadingWrapper
