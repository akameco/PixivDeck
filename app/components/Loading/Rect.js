// @flow
import styled, { keyframes } from 'styled-components'

const loading = keyframes`
	0%,
	40%,
	100% {
		transform: scaleY(0.4);
	}

	20% {
		transform: scaleY(1);
	}
`

const Rect = styled.div`
	background-color: #333;
	height: 100%;
	width: 6px;
	display: inline-block;
	margin: 1px;
	animation: ${loading} 1.2s infinite ease-in-out;
	animation-delay: ${props => props.delay}s;
`

export default Rect
