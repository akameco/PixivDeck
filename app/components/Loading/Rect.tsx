import styled, { keyframes } from 'styled-components'
import { key } from 'styles/styleTheme'

const loading = keyframes`
  0%,
  40%,
  100% {
    transform: scaleY(0.4);
  }

  20% {
    transform: scaleY(1);
  }
`
const Rect = styled.div`
  background-color: ${key('gray')};
  height: 100%;
  width: 6px;
  display: inline-block;
  margin: 1px;
  animation: ${loading} 1.2s infinite ease-in-out;
  animation-delay: ${props => props.delay}s;
`
export default Rect
