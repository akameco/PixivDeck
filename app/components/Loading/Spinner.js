// @flow
import styled from 'styled-components';

const Spinner = styled.div`
	margin: 100px auto;
	width: 50px;
	height: 40px;
	text-align: center;
	font-size: 10px;
`;

export default Spinner;
