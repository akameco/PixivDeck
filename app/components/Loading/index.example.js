// @flow
import * as React from 'react'
import { storiesOf } from '@storybook/react'
import Loading from '.'

storiesOf('Loading', module).add('render', () => {
  return <Loading />
})
