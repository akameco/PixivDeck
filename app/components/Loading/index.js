// @flow
import React from 'react';
import LoadingWrapper from './LoadingWrapper';
import Spinner from './Spinner';
import Rect from './Rect';

type Props = {
  wrapStyle?: Object,
  style?: Object,
};

function Loading({style, wrapStyle}: Props) {
  return (
    <LoadingWrapper style={wrapStyle}>
      <Spinner style={style}>
        <Rect delay={0} />
        <Rect delay={-1.1} />
        <Rect delay={-1} />
        <Rect delay={-0.9} />
        <Rect delay={-0.8} />
      </Spinner>
    </LoadingWrapper>
  );
}

export default Loading;
