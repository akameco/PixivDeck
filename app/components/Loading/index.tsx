import * as React from 'react'
import LoadingWrapper from './LoadingWrapper'
import Spinner from './Spinner'
import Rect from './Rect'

interface Props {
  wrapStyle?: object
  style?: object
}

function Loading({ style, wrapStyle }: Props) {
  return (
    <LoadingWrapper style={wrapStyle}>
      <Spinner style={style}>
        <Rect delay={0} />
        <Rect delay={-1.1} />
        <Rect delay={-1} />
        <Rect delay={-0.9} />
        <Rect delay={-0.8} />
      </Spinner>
    </LoadingWrapper>
  )
}

export default Loading
