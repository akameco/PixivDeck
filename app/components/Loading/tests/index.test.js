import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import Loading from '../';

const renderComponent = (props = {}) => shallow(<Loading {...props} />);

test('snapshot', () => {
  const c = renderComponent();
  expect(shallowToJson(c)).toMatchSnapshot();
});
