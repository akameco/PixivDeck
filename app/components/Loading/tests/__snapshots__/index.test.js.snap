exports[`test snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      delay={0} />
    <styled.div
      delay={-1.1} />
    <styled.div
      delay={-1} />
    <styled.div
      delay={-0.9} />
    <styled.div
      delay={-0.8} />
  </styled.div>
</styled.div>
`;
